mvlearn/ajive/ajive.py

```diff
@@ -0,0 +1,760 @@
+        All singular values from the concatenated joint matrix.
+
+    random_sv_samples_: list
+        Random singular value samples from random direction bound.
+
+    rand_cutoff_: float
+        Singular value squared cutoff for the random direction bound.
+
+    wedin_samples_: dict
+        The wedin samples for each view.
+
+    wedin_cutoff_: float
+        Singular value squared cutoff for the wedin bound.
+
+    svalsq_cutoff_: float
+        max(rand_cutoff_, wedin_cutoff_)
+
+    joint_rank_wedin_est_: int
+        The joint rank estimated using the wedin/random direction bound.
+
+    joint_rank: int
+        The rank of the joint matrix
+
+    indiv_ranks: dict of ints
+        Ranks of the individual matrices for each view.
+
+    """
+
+    def __init__(self,
+                 init_signal_ranks,
+                 joint_rank=None, indiv_ranks=None,
+                 center=True,
+                 reconsider_joint_components=True,
+                 wedin_percentile=5, n_wedin_samples=1000,
+                 precomp_wedin_samples=None,
+                 randdir_percentile=95, n_randdir_samples=1000,
+                 precomp_randdir_samples=None,
+                 store_full=True, n_jobs=None):
+
+        self.init_signal_ranks = init_signal_ranks
+        self.joint_rank = joint_rank
+        self.indiv_ranks = indiv_ranks
+
+        self.center = center
+
+        self.wedin_percentile = wedin_percentile
+        self.n_wedin_samples = n_wedin_samples
+        self.wedin_samples_ = precomp_wedin_samples
+        if precomp_wedin_samples is not None:
+            self.n_wedin_samples = len(list(precomp_wedin_samples.values())[0])
+
+        self.randdir_percentile = randdir_percentile
+        self.n_randdir_samples = n_randdir_samples
+        self.random_sv_samples_ = precomp_randdir_samples
+        if precomp_randdir_samples is not None:
+            self.n_randdir_samples = len(precomp_randdir_samples)
+
+        self.reconsider_joint_components = reconsider_joint_components
+
+        self.store_full = store_full
+
+        self.n_jobs = n_jobs
+
+    def __repr__(self):
+
+        if self.is_fit:
+            r = "joint rank: {}".format(self.common.rank)
+            for bn in self.block_names:
+                indiv_rank = self.blocks[bn].individual.rank
+                r += ", block {} indiv rank: {}".format(bn, indiv_rank)
+            return r
+
+        else:
+            return "No data has been fitted yet"
+
+    def fit(self, blocks, precomp_init_svd=None):
+        """
+        Fits the AJIVE decomposition.
+
+        Parameters
+        ----------
+        blocks: {list of array-likes or numpy.ndarray, dict}
+            - blocks length: n_views
+            - blocks[i] shape: (n_samples, n_features_i) 
+            The different views that are input. Input as data matrices. 
+            If dict, will name blocks by keys, otherwise blocks are named by 
+            0, 1, ...K. 
+            
```

Comment:
Is there a specific reason this is allowed to be a dict, not just a list of array-likes? Does it have something to do with the way ajive is typically used in practice that the input would like be formatted as a dict already?

Reply:
Then return is as a dictionary as there are 3 different matrices returned. Also usually inputs are views with specific names. The keys make it easy to call those views and specify which of the 3 matrices you want

Reply:
Ok that's fine then, is there another way you can write the "type" of each input in the Parameters section though? I was very confused looking at {list, dict} the first time